src/hook/useAppBackgroundApiReconnect.ts

```diff
@@ -0,0 +1,27 @@
+import {useEffect, useRef} from 'react';
+import {useApiReconnect} from 'context/ChainApiContext';
+import {useNavigation} from '@react-navigation/core';
+import {apiLoadingScreen} from 'src/navigation/routeKeys';
+import {useAppState} from 'src/hook/useAppState';
+
+export function useAppBackgroundApiReconnect() {
+  const {canReconnect, reconnect} = useApiReconnect();
+  const navigation = useNavigation();
+  const appState = useAppState();
+  const didAppCameToForegroundOnce = useRef(false);
+
+  useEffect(() => {
+    didAppCameToForegroundOnce.current = appState.didAppCameToForeground;
+  }, [appState]);
+
+  useEffect(() => {
+    if (didAppCameToForegroundOnce.current && canReconnect) {
+      reconnect();
+      setTimeout(() => {
```

Comment:
why do we need this?

Reply:
Once the api disconnects the app tries to navigate to `connectionRetryScreen` from `ChainApiContext` and to `apiLoadingScreen` from here.
On android this back to back navigation doesn't work properly.